apps/prairielearn/src/pages/instructorAssessmentInstances/instructorAssessmentInstances.js

```diff
@@ -1,191 +1,204 @@
 const debug = require('debug')('prairielearn:' + path.basename(__filename, '.js'));
-import * as util from 'node:util';
+import { z } from 'zod';
 
 import * as error from '@prairielearn/error';
 import { regradeAssessmentInstance } from '../../lib/regrading';
 import {
-  checkBelongs,
-  gradeAssessmentInstance,
+  checkBelongsAsync,
+  gradeAssessmentInstanceAsync,
   gradeAllAssessmentInstances,
 } from '../../lib/assessment';
 import * as sqldb from '@prairielearn/postgres';
+import { IdSchema } from '../../lib/db-types';
 
 const router = express.Router();
 const sql = sqldb.loadSqlEquiv(__filename);
 
-router.get('/raw_data.json', function (req, res, next) {
-  debug('GET /raw_data.json');
-  if (!res.locals.authz_data.has_course_instance_permission_view) {
-    return next(error.make(403, 'Access denied (must be a student data viewer)'));
-  }
-  const params = {
-    assessment_id: res.locals.assessment.id,
-    group_work: res.locals.assessment.group_work,
-  };
-  sqldb.query(sql.select_assessment_instances, params, function (err, result) {
-    if (ERR(err, next)) return;
-    res.send(result.rows);
-    return;
-  });
+const assessmentInstanceRowSchema = z.object({
+  assessment_label: z.string(),
+  user_id: IdSchema.nullable(),
+  uid: z.string().nullable(),
+  name: z.string().nullable(),
+  role: z.string().nullable(),
+  group_id: IdSchema.nullable(),
+  group_name: z.string().nullable(),
+  uid_list: z.array(z.string()).nullable(),
+  user_name_list: z.array(z.string()).nullable(),
+  group_roles: z.array(z.string()).nullable(),
+  username: z.string().nullable(),
+  score_perc: z.number().nullable(),
+  points: z.number(),
+  max_points: z.number(),
+  number: z.number(),
+  assessment_instance_id: IdSchema,
+  open: z.boolean(),
+  time_remaining: z.string(),
+  time_remaining_sec: z.number().nullable(),
+  total_time: z.string(),
+  total_time_sec: z.number().nullable(),
+  date: z.date(),
+  date_formatted: z.string(),
+  duration: z.string(),
+  duration_secs: z.number(),
+  duration_mins: z.number(),
+  highest_score: z.boolean(),
+  client_fingerprint_id_change_count: z.number(),
 });
 
-router.get('/client.js', function (req, res, next) {
-  debug('GET /client.js');
-  if (!res.locals.authz_data.has_course_instance_permission_view) {
-    return next(error.make(403, 'Access denied (must be a student data viewer)'));
-  }
-  res.type('text/javascript');
-  res.render(__filename.replace(/\.js$/, 'ClientJS.ejs'), res.locals);
-});
+router.get(
+  '/raw_data.json',
+  asyncHandler(async (req, res) => {
+    if (!res.locals.authz_data.has_course_instance_permission_view) {
+      throw error.make(403, 'Access denied (must be a student data viewer)');
+    }
+    const result = await sqldb.queryRows(
+      sql.select_assessment_instances,
+      {
+        assessment_id: res.locals.assessment.id,
+        group_work: res.locals.assessment.group_work,
+      },
+      assessmentInstanceRowSchema,
```

Comment:
Schema names should be capitalized.

Reply:
This schema name is now capitalized in commit [8472ae8](https://github.com/PrairieLearn/PrairieLearn/pull/9080/commits/8472ae8a0869182f87c71e15b748cbd6064db927).